docs/tutorials/clojure-guide.md

```diff
@@ -0,0 +1,238 @@
+
+Once it is running (and you are plugged into the REPL), `CIDER` will provide many of the same features that `clojure-lsp` does, so you need to tell Emacs how to resolve conflicts.
+
+Therefore, if you want to have a more *IDE-ish* experience with Emacs, you should read this guide.
+
+## Basic configuration
+
+Here is a bare-bones `lsp-mode` configuration template to get you started with your own `lsp-mode` config, 
+or to try out in a separate one-off session. Please note that Emacs configuration frameworks such as 
+**Spacemacs** or **Doom Emacs** often ship with `lsp-mode` settings of their own; should you be using such a framework, 
+and find that `lsp-mode` doesn't behave as intended, please make sure to follow this tutorial from a clean starting point.
+in your config or you could run in a separate session.
+
+``` emacs-lisp
+(require 'package)
+(add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/") t)
+(package-initialize)
+
+(setq package-selected-packages '(clojure-mode lsp-mode cider yasnippet lsp-treemacs helm-lsp projectile hydra flycheck company avy which-key helm-xref))
+
+(when (cl-find-if-not #'package-installed-p package-selected-packages)
+  (package-refresh-contents)
+  (mapc #'package-install package-selected-packages))
+
+;; sample `helm' configuration use https://github.com/emacs-helm/helm/ for details
+(helm-mode)
+(require 'helm-xref)
+(define-key global-map [remap find-file] #'helm-find-files)
+(define-key global-map [remap execute-extended-command] #'helm-M-x)
+(define-key global-map [remap switch-to-buffer] #'helm-mini)
+
+(which-key-mode)
+(add-hook 'clojure-mode-hook 'lsp)
+(add-hook 'clojurescript-mode-hook 'lsp)
+(add-hook 'clojurec-mode-hook 'lsp)
+
+(setq gc-cons-threshold (* 100 1024 1024)
+      read-process-output-max (* 1024 1024)
+      treemacs-space-between-root-nodes nil
+      company-idle-delay 0.0
+      company-minimum-prefix-length 1
+      lsp-lens-enable t
+      lsp-signature-auto-activate nil 
+      ; lsp-enable-indentation nil ; uncomment to use cider indentation instead of lsp
+      ; lsp-enable-completion-at-point nil ; uncomment to use cider completion instead of lsp
+      )
+
+(with-eval-after-load 'lsp-mode
+  (add-hook 'lsp-mode-hook #'lsp-enable-which-key-integration)
+  (yas-global-mode))
+```
+
+For more detailed info on how to setup `lsp-mode`, check [here](https://emacs-lsp.github.io/lsp-mode/page/installation/).
+
+## Installing the language server
+
+`lsp-mode` is a client-server application with `Emacs` acting as the client. In
+order for it to work, you have to install a separate _language server_ that
+understands the specifics of your target language. 
+
+For Clojure, we will use [clojure-lsp](https://github.com/clojure-lsp/clojure-lsp) which as of 2021/01, is the only [LSP](https://microsoft.github.io/language-server-protocol/implementors/servers/) 
+server for Clojure, it has a lot of useful features that I will try to show in this guide.
```

Comment:
"it has" or "and has"? The end of this sentence seems wrong to me.

Reply:
Not sure, WDYT @leungbk ?

Reply:
I like "and has".

Reply:
```suggestion
server for Clojure, and has a lot of useful features that I will try to show in this guide.
```